pkg/controller/handlers/cleanup/user.go

```diff
@@ -0,0 +1,57 @@
+package cleanup
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/obot-platform/nah/pkg/router"
+	gclient "github.com/obot-platform/obot/pkg/gateway/client"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type UserCleanup struct {
+	gatewayClient *gclient.Client
+}
+
+func NewUserCleanup(gatewayClient *gclient.Client) *UserCleanup {
+	return &UserCleanup{
+		gatewayClient: gatewayClient,
+	}
+}
+
+func (u *UserCleanup) Cleanup(req router.Request, _ router.Response) error {
+	userDelete := req.Object.(*v1.UserDelete)
+	var projects v1.ThreadList
```

Comment:
Nit: `Threads` might be more appropriate for variable name

Reply:
Changed.